SPEC.md

```diff
@@ -0,0 +1,675 @@
+| **claimBlock**           | The block during which the claim was created.                                                 |
+| **endVerificationBlock** | The block at which the verification period for the claim is complete.                         |
+| **endSlashingBlock**     | The block at which the slashing period for the claim is complete.                             |
+| **segmentVerifications** | The segments that have been submitted for verification.                                       |
+| **status**               | The status of the claim (pending, slashed, complete)                                          |
+
+The **claimRoot** is computed as follows:
+
+Given segments `s_1, s_2,..., s_n`, the transcoder creates transcode receipts `r_1, r_2,..., r_n`. The transcoder computes the transcode receipt hashes
+`keccak256(r_1), keccak256(r_2),..., keccak256(r_n)`. Then, the transcoder creates a Merkle tree using the transcode receipt hashes as the leaves of the tree. The
+root of the Merkle tree is used as **claimRoot** in a transcode claim.
+
+#### Transcode Proof
+
+Transcoders submit transcode proofs for claimed segments that are challenged for verification.
+
+| Field               | Description                                                                                                                              |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| **jobId**           | The transcode job identifier.                                                                                                            |
+| **claimId**         | The transcode claim identifier.                                                                                                          |
+| **segmentNumber**   | The segment's position in the sequential ordering of segments in the stream.                                                             |
+| **dataStorageHash** | The content-addressed storage hash used to retrieve the segment data payload from a distributed storage network (used for verification). |
+| **dataHashes**      | The Keccak-256 hash of the input segment data payload and the Keccak-256 hash of the transcoded segment data payload.                    |
+| **broadcasterSig**  | The broadcaster's signature over the segment hash `keccak256(streamId, segmentNumber, dataHash)`.                                        |
+| **receiptProof**    | The Merkle proof proving the transcode receipt was submitted with the Merkle root commitment contained in the transcode claim.           |
+
+The corresponding transcode receipt hash for **receiptProof** can be computed as follows:
+
+Let `jobs` be a mapping of job IDs to transcode jobs. The transcode receipt hash is `keccak256(jobs[jobId].streamId, segmentNumber, dataHashes[0], dataHashes[1], broadcasterSig)`.
+
+#### Transcoder Pool
```

Comment:
This section might be just a little too specific in the implementation details. It could be simplified to one or two sentences probably.